More type-hints and documentation added

Quix0r opened the pull request:
This PR contains more added type-hints, 1 hour straight no `TypeError` or anything related to these changes.

Quix0r committed:
Changes:
- added missing type-hints
- added missing documentation

Quix0r committed:
Changes:
- added more type-hints
- added missing documentation

Quix0r committed:
Continued:
- added more type-hints
- added some missing documentation
- Return the result from DBA::delete() to let other methods know about it

Quix0r committed:
Continued:
- changed some double-quotes to single
- added missing type-hints
- added missing documentation
- fixed indenting a bit

Quix0r committed:
Renamed variable, no need for "orig_" prefix

Quix0r committed:
Added missing type-hints

Quix0r committed:
Nodeinfo::getOrganization() doesn't need configuration object being inserted
when you have DI::config() around.

Quix0r committed:
Changes:
- added more type-hints
- added missing documentation

Quix0r committed:
Maybe fix for:

"Argument 1 passed to Friendica\Model\ItemURI::getIdByURI() must be of the type string, null given, called in Processor.php line 1219"

Quix0r committed:
Proper type is string

MrPetovan commented on src/Model/Item.php:
```diff
@@ -342,7 +343,7 @@ public static function markForDeletionById($item_id, $priority = PRIORITY_HIGH)
 		Post\Category::storeTextByURIId($item['uri-id'], $item['uid'], '');
 
-		if (!Post::exists(["`uri-id` = ? AND `uid` != 0 AND NOT `deleted`", $item['uri-id']])) {
+		if (!Post::exists(['`uri-id` = ? AND `uid` != 0 AND NOT `deleted`', $item['uri-id']])) {
```
The convention is to use double-quote for SQL statements or conditions.
```suggestion
		if (!Post::exists(["`uri-id` = ? AND `uid` != 0 AND NOT `deleted`", $item['uri-id']])) {
```

MrPetovan commented on src/Model/Item.php:
```diff
@@ -497,7 +511,7 @@ private static function isDuplicate(array $item)
 		} elseif ($item['network'] == Protocol::OSTATUS) {
 			// Check for an existing post with the same content. There seems to be a problem with OStatus.
-			$condition = ["`body` = ? AND `network` = ? AND `created` = ? AND `contact-id` = ? AND `uid` = ?",
+			$condition = ['`body` = ? AND `network` = ? AND `created` = ? AND `contact-id` = ? AND `uid` = ?',
```
```suggestion
			$condition = ["`body` = ? AND `network` = ? AND `created` = ? AND `contact-id` = ? AND `uid` = ?",
```

Quix0r replied:
Do you want me to fix the indenting here?

MrPetovan replied:
No, it's fine, the values are following the SQL snippet order, it's better this way.

MrPetovan commented on src/Model/Item.php:
```diff
@@ -623,15 +637,20 @@ public static function isTooOld(array $item)
-			$condition = ["`uri-id` = ? AND `uid` = ? AND `network` IN (?, ?, ?, ?)",
-				$item['uri-id'], $item['uid'],
-				Protocol::ACTIVITYPUB, Protocol::DIASPORA, Protocol::DFRN, Protocol::OSTATUS];
+			$condition = ['`uri-id` = ? AND `uid` = ? AND `network` IN (?, ?, ?, ?)',
```
```suggestion
			$condition = ["`uri-id` = ? AND `uid` = ? AND `network` IN (?, ?, ?, ?)",
```

Quix0r replied:
You have `$condition = ['`` on more places:
- PostUpdate.php, lines 140, 347, 1441
- OpenWebAuthToken.php, line 87
- Item.php, line 646
- Contact.php, lines 274, 995
- DRN.php, lines 230, 2324

Do you want me to fix them all?

MrPetovan replied:
No thank you, I'd rather you only focus on the files you've already edited.

MrPetovan commented on src/Model/Nodeinfo.php:
```diff
@@ -156,12 +156,16 @@ public static function getServices()
 
-		if (!empty($config->get('config', 'admin_email'))) {
-			$adminList = explode(',', str_replace(' ', '', $config->get('config', 'admin_email')));
+		if (!empty(DI::config()->get('config', 'admin_email'))) {
+			$adminList = explode(',', str_replace(' ', '', DI::config()->get('config', 'admin_email')));
```
Please keep the explicit dependency, `DI::config()` is a temporary crutch for the legacy static `Config` but it shouldn't replace an actual object.
```suggestion
		if (!empty($config()->get('config', 'admin_email'))) {
			$adminList = explode(',', str_replace(' ', '', $config()->get('config', 'admin_email')));
```

MrPetovan commented on src/Model/Nodeinfo.php:
```diff
@@ -156,12 +156,16 @@ public static function getServices()
 	}
 
-	public static function getOrganization($config)
+	public static function getOrganization()
```
```suggestion
	public static function getOrganization(\Friendica\Core\Config\Capability\IManageConfigValues $config)
```

Quix0r replied:
I usually like to have these long class names (with namespace) in top of the file with a proper `use` statement and then have the source code smaller.

MrPetovan replied:
Of course, but I can't make fancy code suggestions like these. Feel free to add the namespace in a `use` statement.

Quix0r replied:
Fixed this + missing doctag and return type.

MrPetovan commented on src/Model/OpenWebAuthToken.php:
```diff
@@ -82,10 +82,10 @@ public static function getMeta($type, $uid, $token)
 	{
-		$condition = ["`type` = ? AND `created` < ?", $type, DateTimeFormat::utcNow() . " - INTERVAL " . $interval];
-		DBA::delete("openwebauth-token", $condition);
+		$condition = ['`type` = ? AND `created` < ?', $type, DateTimeFormat::utcNow() . ' - INTERVAL ' . $interval];
```
```suggestion
		$condition = ["`type` = ? AND `created` < ?", $type, DateTimeFormat::utcNow() . ' - INTERVAL ' . $interval];
```

MrPetovan commented on src/Model/Photo.php:
```diff
@@ -156,19 +156,19 @@ public static function getPhotoForUser($uid, $resourceid, $scale = 0, array $con
 
-		$conditions = ["`resource-id` = ? AND `scale` <= ? " . $sql_acl, $resourceid, $scale];
-		$params = ["order" => ["scale" => true]];
+		$conditions = ['`resource-id` = ? AND `scale` <= ? ' . $sql_acl, $resourceid, $scale];
```
```suggestion
		$conditions = ["`resource-id` = ? AND `scale` <= ? " . $sql_acl, $resourceid, $scale];
```

MrPetovan commented on src/Model/Photo.php:
```diff
@@ -331,71 +331,71 @@ public static function createPhotoForExternalResource($url, $uid = 0, $mimetype
+	public static function store(Image $image, int $uid, int $cid, string $rid, string $filename, string $album, int $scale, int $type = self::DEFAULT, string $allow_cid = '', string $allow_gid = '', string $deny_cid = '', string $deny_gid = '', string $desc = ''): bool
 	{
-		$photo = self::selectFirst(["guid"], ["`resource-id` = ? AND `guid` != ?", $rid, ""]);
+		$photo = self::selectFirst(['guid'], ['`resource-id` = ? AND `guid` != ?', $rid, '']);
```
```suggestion
		$photo = self::selectFirst(['guid'], ["`resource-id` = ? AND `guid` != ?", $rid, '']);
```

MrPetovan commented on src/Model/Photo.php:
```diff
@@ -331,71 +331,71 @@ public static function createPhotoForExternalResource($url, $uid = 0, $mimetype
 	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
 	 */
-	public static function store(Image $Image, $uid, $cid, $rid, $filename, $album, $scale, $type = self::DEFAULT, $allow_cid = "", $allow_gid = "", $deny_cid = "", $deny_gid = "", $desc = "")
+	public static function store(Image $image, int $uid, int $cid, string $rid, string $filename, string $album, int $scale, int $type = self::DEFAULT, string $allow_cid = '', string $allow_gid = '', string $deny_cid = '', string $deny_gid = '', string $desc = ''): bool
```
I prefer to capitalize variables holding structured objects as they are in minority and I feel like they should stand out.
```suggestion
	public static function store(Image $Image, int $uid, int $cid, string $rid, string $filename, string $album, int $scale, int $type = self::DEFAULT, string $allow_cid = '', string $allow_gid = '', string $deny_cid = '', string $deny_gid = '', string $desc = ''): bool
```

Quix0r replied:
No, let's make all variables the same, please no extra rules. :-)

MrPetovan replied:
How do you feel about it, @annando  and @nupplaphil ?

MrPetovan commented on src/Model/Photo.php:
```diff
@@ -433,63 +433,63 @@ public static function delete(array $conditions, array $options = [])
 	 * @see   \Friendica\Database\DBA::update
 	 */
-	public static function update($fields, $conditions, Image $img = null, array $old_fields = [])
+	public static function update(array $fields, array $conditions, Image $image = null, array $old_fields = []): bool
```
```suggestion
	public static function update(array $fields, array $conditions, Image $Image = null, array $old_fields = []): bool
```

MrPetovan commented on src/Model/Photo.php:
```diff
@@ -507,66 +507,66 @@ public static function importProfilePhoto($image_url, $uid, $cid, $quit_on_error
-		$Image = new Image($img_str, $type);
-		if ($Image->isValid()) {
-			$Image->scaleToSquare(300);
+		$image = new Image($img_str, $type);
```
```suggestion
		$Image = new Image($img_str, $type);
```

MrPetovan commented on src/Model/Photo.php:
```diff
@@ -590,31 +590,33 @@ public static function importProfilePhoto($image_url, $uid, $cid, $quit_on_error
 	 * @return float
 	 */
-	public static function getGps($exifCoord, $hemi)
+	public static function getGps(array $exifCoord, strinf $hemi): float
```
Typo.
```suggestion
	public static function getGps(array $exifCoord, string $hemi): float
```

MrPetovan commented on src/Model/Photo.php:
```diff
@@ -885,17 +889,23 @@ public static function isLocalPage($name)
+	 * @param Image $image Image instance
+	 * @return Image|null Image instance on success, null on error
+	 */
+	private static function fitImageSize(Image $image)
```
```suggestion
	private static function fitImageSize(Image $Image)
```

MrPetovan commented on src/Model/Photo.php:
```diff
@@ -939,17 +955,23 @@ private static function loadImageFromURL(string $image_url)
 		$type = Images::getMimeTypeByData($img_str, $image_url, $type);
 
-		$Image = new Image($img_str, $type);
+		$image = new Image($img_str, $type);
```
```suggestion
		$Image = new Image($img_str, $type);
```

MrPetovan commented on src/Model/Photo.php:
```diff
@@ -1008,34 +1030,36 @@ private static function uploadImage(array $files)
 		$imagedata = @file_get_contents($src);
-		$Image = new Image($imagedata, $filetype);
-		if (!$Image->isValid()) {
+		$image = new Image($imagedata, $filetype);
```
```suggestion
		$Image = new Image($imagedata, $filetype);
```

MrPetovan commented on src/Model/Photo.php:
```diff
@@ -1052,10 +1076,10 @@ public static function upload(int $uid, array $files, string $album = '', string
 		}
 
-		$Image    = $data['image'];
+		$image    = $data['image'];
```
```suggestion
		$Image    = $data['image'];
```

MrPetovan commented on src/Model/Photo.php:
```diff
@@ -1139,10 +1163,10 @@ public static function uploadAvatar(int $uid, array $files, string $url = ''): s
 		}
 
-		$Image    = $data['image'];
+		$image    = $data['image'];
```
```suggestion
		$Image    = $data['image'];
```

MrPetovan commented on src/Model/Photo.php:
```diff
@@ -1217,19 +1241,19 @@ public static function uploadBanner(int $uid, array $files = [], string $url = '
 		}
 
-		$Image    = $data['image'];
+		$image    = $data['image'];
```
```suggestion
		$Image    = $data['image'];
```

MrPetovan commented on src/Module/NodeInfo210.php:
```diff
@@ -57,7 +57,7 @@ protected function rawContent(array $request = [])
 				'version'  => FRIENDICA_VERSION . '-' . DB_UPDATE_VERSION,
 			],
-			'organization'      => Nodeinfo::getOrganization($this->config),
+			'organization'      => Nodeinfo::getOrganization(),
```
```suggestion
			'organization'      => Nodeinfo::getOrganization($this->config),
```

Quix0r committed:
Changes:
- added type-hints
- added documentation
- changed double-quotes to single

Quix0r committed:
Naming-convention:
- variables should start lower-case: $image

Quix0r committed:
Changed back to suggestions by @MrPetovan

Quix0r committed:
Changed to suggestings (back to original) + fixed typo in scalar type

Quix0r committed:
Ops, also this!

Quix0r committed:
Changes:
- changed to proper "use Foo\Bar;"
- added doctag
- added return type as this is fixed

Quix0r committed:
Changes:
- added more type-hints
- DBStructure::existsTable() does no longer need array support for table name
  because this is no longer used (good work!)

Quix0r committed:
Wrong returned type, has to be array

Quix0r committed:
Changes:
- dbstructure(null) is no longer possible, an empty string does it

Quix0r committed:
Contact::getAccountType()'s parameter is never a string, ops

Quix0r committed:
Changes:
- MySQL index on BLOB/TEXT can only be partial (e.g.: `column`(length))

Quix0r committed:
Changes:
- added missing type-hints
- added documentation for a method

Quix0r committed:
Some calls saved

MrPetovan commented:
From Woodpecker:
> ```
> 1) Friendica\Test\src\Module\Api\Friendica\NotificationTest::testWithXmlResult
> Failed asserting that two DOM documents are equal.
> --- Expected
> +++ Actual
> @@ @@
> 
> -
> +
> 
> /woodpecker/src/github.com/friendica/friendica/tests/src/Module/Api/Friendica/NotificationTest.php:72
> ```

Quix0r commented:
Same problem here. I don't know what have caused this.

Quix0r committed:
Changes:
- added type-hints
- added returned type-hints in interface (I checked all)

Quix0r committed:
Ops, not here wanted. :-(

Quix0r commented:
@MrPetovan @nupplaphil @annando Do you know how to fix that error coming from the unit test?

Quix0r committed:
Incremented database version
